packages/@azure/keyvault/lib/secretsClient.ts

```diff
@@ -345,6 +345,78 @@ export class SecretsClient {
 
     return asyncGenerator();
   }
+  
+  /**
+   * Iterates the latest version of all secrets in the vault.  The full secret identifier and attributes are provided 
+   * in the response. No values are returned for the secrets. This operations requires the secrets/list permission.
+   * @summary List all versions of the specified secret.
+   * @param secretName The name of the secret.
+   * @param [options] The optional parameters
+   * @returns AsyncIterableIterator<Secret>
+   */
+  public getAllSecrets(
+    options?: RequestOptionsBase
+  ): AsyncIterableIterator<Secret> {
+    const keyVaultClient = this.client;
+    const keyVaultUrl = this.vaultBaseUrl;
+    const extractSecretFromSecretItem = this.getSecretFromSecretBundle;
+
+    async function* asyncGenerator() {
+      let currentSetResponse = await keyVaultClient.getSecrets(
+        keyVaultUrl,
+        options
+      );
+      let currentSetLength = currentSetResponse.length;
+      let i = 0;
+      while (i < currentSetLength) {
+        yield extractSecretFromSecretItem(currentSetResponse[i]);
+        i++;
+        if (i === currentSetLength && currentSetResponse.nextLink) {
+          currentSetResponse = await keyVaultClient.getSecretsNext(currentSetResponse.nextLink, options);
+          i = 0;
+          currentSetLength = currentSetResponse.length;
+        }
+      }
+    }
+
+    return asyncGenerator();
+  }

```

Comment:
As discussed in the other commit. This method can be defined as

```typescript
  public async *getAllSecrets(
    options?: RequestOptionsBase
  ): AsyncIterableIterator<Secret> {
    const keyVaultClient = this.client;
    const keyVaultUrl = this.vaultBaseUrl;
    const extractSecretFromSecretItem = this.getSecretFromSecretBundle;

    let currentSetResponse = await keyVaultClient.getSecrets(
      keyVaultUrl,
      options
    );
    yield* currentSetResponse.map(extractSecretFromSecretItem);

    while (currentSetResponse.nextLink) {
      currentSetResponse = await keyVaultClient.getSecretVersionsNext(currentSetResponse.nextLink, options);
      yield* currentSetResponse.map(extractSecretFromSecretItem);
    }
  }
```

Reply:
I'll take care of that in a separate commit

Reply:
I like the start `public async* getAllSecrets` personally (star next to async rather than the name).